src/lib/hash/ascon_hash256/ascon_hash256.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+* Ascon-Hash256 (NIST SP.800-232)
+* (C) 2025 Jack Lloyd
+*.    2025 René Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/assert.h>
+#include <botan/internal/ascon_hash256.h>
+
+namespace Botan {
+
+Ascon_Hash256::Ascon_Hash256() {
+   init();
+}
+
+std::unique_ptr<HashFunction> Ascon_Hash256::new_object() const {
+   return std::make_unique<Ascon_Hash256>();
+}
+
+std::unique_ptr<HashFunction> Ascon_Hash256::copy_state() const {
+   return std::make_unique<Ascon_Hash256>(*this);
+}
+
+void Ascon_Hash256::init() {
+   // See NIST SP.800-232, Section 5.1.1 (Initialization)
+   m_ascon_p.clear();
+   m_ascon_p.permute<12>();
```

Comment:
Note to self: I'm not convinced anymore that passing the IV as a template parameter was that smart after all. Probably better to just pass it here and statically make sure that the right amount of IV words is passed.

Or: actually make all the parameters that build up the IV explicit and calculate it constexpr as explained in SP.800-232 Apx. B. That might be a liiiiittle overkill, admittedly. 😏

Reply:
Introduced an `AsconParameters` struct that holds these values as specified in Appendix B.